From 18b2d00f1077720e2520f925db0a4402681633d3 Mon Sep 17 00:00:00 2001
From: Alison Wang <alison.wang@nxp.com>
Date: Thu, 8 Mar 2018 11:43:49 +0800
Subject: [PATCH 397/767] Revert "drm/fsl-dcu: Fix the interrupt issue in
 suspend/resume functions"

This reverts commit 9e3a08b475e6445cb9f29c1278d91a9464848edf.

As upstream commit bacbe44889828852678bc74ad0e942a7684f206f
"drm/fsl-dcu: enable IRQ before drm_atomic_helper_resume()" can also fix
the original issue, the commit 9e3a08b475e6445cb9f29c1278d91a9464848edf
needs to be reverted.

Signed-off-by: Alison Wang <alison.wang@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.09.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/gpu/drm/fsl-dcu/fsl_dcu_drm_drv.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/fsl-dcu/fsl_dcu_drm_drv.c b/drivers/gpu/drm/fsl-dcu/fsl_dcu_drm_drv.c
index 50490f4fd0f5..3447c1c0a4de 100644
--- a/drivers/gpu/drm/fsl-dcu/fsl_dcu_drm_drv.c
+++ b/drivers/gpu/drm/fsl-dcu/fsl_dcu_drm_drv.c
@@ -194,14 +194,14 @@ static int fsl_dcu_drm_pm_suspend(struct device *dev)
 	if (!fsl_dev)
 		return 0;
 
+	disable_irq(fsl_dev->irq);
+
 	ret = drm_mode_config_helper_suspend(fsl_dev->drm);
 	if (ret) {
 		enable_irq(fsl_dev->irq);
 		return ret;
 	}
 
-	disable_irq(fsl_dev->irq);
-
 	clk_disable_unprepare(fsl_dev->clk);
 
 	return 0;
-- 
2.17.0

